Display version of BaGetter in a footer on Website

FroggieFrog opened the pull request:
This PR adds a footer to the Website displaying the current running version of BaGetter.
This fixes a part of the requests from #102.

Current state:

![grafik](https://github.com/bagetter/BaGetter/assets/6222752/6ce91d90-3f78-42eb-9755-97ed481d07f1)

**Todo**

Currently there is no version assigned to the binaries of BaGetter. So `1.0.0.0` is always shown.

- [x] add version to BaGetter -> this is done by PR #113

seriouz commented:
Yes, I have a similar view to @Regenhardt . A small footer at the bottom of the page with only version X.Y.Z of the release. And later, when we have the stats page, we'll include the commit hash there (see issue https://github.com/bagetter/BaGetter/issues/102).

loic-sharma commented:
FYI, nuget.org pages all have an HTML comment at the bottom:

```
<!--
    This is the NuGet Gallery version 4.4.5-main.
        Deployed from 58f2934 Link: https://www.github.com/NuGet/NuGetGallery/commit/58f2934
        Built on main Link: https://www.github.com/NuGet/NuGetGallery/tree/main
        Built on 2024-03-07T17:49:07.0000000+00:00
    Deployment label: PROD-USNC.ASE-4.4.5-main-9205298
    You are on AW0MDWK000029.
-->
```

This gives you all the debugging information without any visual clutter.

Regenhardt commented:
> FYI, nuget.org pages all have an HTML comment at the bottom:
> 
> ```
> <!--
>     This is the NuGet Gallery version 4.4.5-main.
>         Deployed from 58f2934 Link: https://www.github.com/NuGet/NuGetGallery/commit/58f2934
>         Built on main Link: https://www.github.com/NuGet/NuGetGallery/tree/main
>         Built on 2024-03-07T17:49:07.0000000+00:00
>     Deployment label: PROD-USNC.ASE-4.4.5-main-9205298
>     You are on AW0MDWK000029.
> -->
> ```
> 
> This gives you all the debugging information without any visual clutter.

Ha that' actually a nice idea. There's only developers using it anyway so that's not overly complicated.

Regenhardt commented on src/BaGetter.Web/Pages/Shared/_Layout.cshtml:
```diff
@@ -60,9 +60,19 @@
+
+    <footer class="footer">        
+        <p>
+            Version: @(Assembly.GetEntryAssembly()
```
Could also add all that git information to that potential version object.

FroggieFrog commented:
Good suggestions. I see what I can come up with.

FroggieFrog commented:
Things I adjusted:
- added a static helper with all version information (`ApplicationVersionHelper`)
- added `<meta>` tags
- added a comment with the version information
- adjusted the `copyright` year
- adjusted the `author` tag
- all version information is set during build (see `Directory.Build.targets`)

some screenshots:

![grafik](https://github.com/bagetter/BaGetter/assets/6222752/28f03484-6146-4cb9-a8a5-44fcef3b278b)

![grafik](https://github.com/bagetter/BaGetter/assets/6222752/28453284-8017-43bc-8643-f2daa402d0f8)

seriouz commented on src/Directory.Build.props:
```diff
@@ -3,7 +3,7 @@
   <PropertyGroup>
     <Authors>Community https://github.com/bagetter/BaGetter; Loic Sharma</Authors>
-    <Copyright>Copyright (c) BaGetter 2023</Copyright>
+    <Copyright>Copyright (c) BaGetter $([System.DateTime]::Now.ToString(yyyy))</Copyright>
```
Ahh i like this so much :D

seriouz commented:
Nice this looks very good!
